src/Stringy.php

```diff
@@ -477,8 +477,21 @@ public function htmlDecode($flags = ENT_COMPAT)
      * @return Stringy  Object with the resulting $str after being html encoded.
      */
     public function htmlEncode($flags = ENT_COMPAT)
-    {
-        $str = htmlentities($this->str, $flags, $this->encoding);
+    {        
+        if (function_exists('html_entity_decode')) {
```

Comment:
Yes! When PHP version is lower than `4.3.0` the `else` block will work as a polyfill.